### DISCUSSION TITLE
ascii2nc: MET/11.1.0 and METplus/5.1.0 to support multiple variables (Python)

### DISCUSSION THREAD

**Discussion by pratiman-91** [dtcenter/METplus#2686] [File I/O] [answered]
Hi,
I am trying to create a netcdf similar to pb2nc with multiple variables using `python`. However, according to documentation (https://met.readthedocs.io/en/latest/Users_Guide/reformat_point.html#ascii2nc-output), `"In this case, the global variable “use_var_id” does not exist or set to false (use_var_id = “false” ;). Three variables (obs_var, obs_units, and obs_desc) related with variable names are not added."` 
If I am understanding it correctly, `ascii2nc` does not support multiple variables. 

 Kindly let me know if something like this exists within MET/METplus. Thanks!

**Comment by DanielAdriaansen** [comment, marked as answer]
Hi @pratiman-91!

Thanks for your question. The netCDF file for point observations generated by `PB2NC` and other MET tools is specific to MET and very complex in its construction. I would strongly advise against attempting to generate this type of file from scratch.

Instead, you can use Python embedding with `ASCII2NC` to get the same result. `ASCII2NC` can support multiple variables via the `var` column in the 11-column data you provide from Python. Here is an example use case of using Python embedding with `ASCII2NC`:
https://metplus.readthedocs.io/en/latest/generated/met_tool_wrapper/ASCII2NC/ASCII2NC_python_embedding.html#sphx-glr-generated-met-tool-wrapper-ascii2nc-ascii2nc-python-embedding-py

Let me know if you have further questions!
